client/core/core.go

```diff
@@ -4078,6 +4338,15 @@ func (c *Core) connectDEX(acctInfo *db.AccountInfo) (*dexConnection, error) {
 		return nil, err
 	}
 
+	var dexPubKey *secp256k1.PublicKey
+	if len(dexCfg.DEXPubKey) > 0 {
+		var err error
+		dexPubKey, err = secp256k1.ParsePubKey(dexCfg.DEXPubKey)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding public key: %w", err)
+		}
+	}
```

Comment:
I've chosen to use the server's pubkey as the source of derivation indices rather than the host name. See also #433. This means that we needed the pubkey before the `'register'` request, but we were returning the pubkey in the `'register'` response. I've added the pubkey to the `'config'` response as well now, and we just handle old servers that don't have this change yet as legacy/imported accounts.